test_genie_python_dae.py

```diff
@@ -594,3 +598,115 @@ def _wait_for_dae_period_change(self, expected_value, get_function):
             else:
                 sleep(1)
         self.fail("dae period or number of periods read timed out")
+
+    def test_GIVEN_x_seconds_have_elapsed_since_start_WHEN_getting_time_since_start_without_pause_THEN_seconds_returned_is_correct(self):
+        """
+        Checks if the seconds elapsed since the start is the same as the expected elapsed seconds.
+        """
+        # Time interval between begin, pause and resume
+        sleep_time = 5
+
+        # Calculating execution time of begin() to add to expected runtime
+        begin_runtime = get_execution_time(g.begin)
+
+        # Running begin() before ending
+        sleep(sleep_time)
+
+        actual = g.get_time_since_start()
+
+        # Calculating expected elapsed time it takes from start
+        expected = sleep_time + begin_runtime
+
+        # Taking the fluctuation of actual runtime into account and tolerating up to 1 sec difference
+        self.assertTrue(abs(expected - actual) < 1)
```

Comment:
I found the test was very temperamental. I think the issue is that you're assuming that the data collection starts as soon as you call begin and all the time that begin takes is running time. I think you can do something like the following to measure from when the run is actually started:
```suggestion
        """
        Checks if the seconds elapsed since the start is the same as the expected elapsed seconds.
        """
        sleep_time = 5

        def get_time_thread(return_value):
            """
            Runs on a thread to start timing as we enter the Running state.
            The return values are appended to the provided list to pass them between threads.
            """
            g.waitfor_runstate("RUNNING")
            sleep(sleep_time)
            return_value.append(g.get_time_since_start())
            return_value.append(g.get_time_since_start(True))

        time_taken = []
        thread = Thread(target=get_time_thread, args=(time_taken,))
        thread.start()

        g.begin()

        thread.join()

        # Taking the fluctuation of actual runtime into account and tolerating up to 1 sec difference
        self.assertAlmostEqual(sleep_time, time_taken[0], delta=1)
        self.assertTrue(abs(timedelta(seconds=sleep_time) - time_taken[1]) < timedelta(seconds=1))
```

I also modified what you had in a couple of other ways:
* `assertAlmostEqual` is nicer as if it fails it will print what the two values are, rather than just True/False
* I think we can get away with testing both forms of the function in one test

If you think this is better can you do the same for the other test and remove the helper function for getting the execution time? Happy to discuss though.